### DISCUSSION TITLE
Expression Editor Window Suggestion (or bug?)

### DISCUSSION THREAD

**Discussion by rminder** [TabularEditor/TabularEditor3#1044] [Ideas]
When I click on a measure I get the Expression Editor window appearing, as I would expect. However, when I click on another measure, I don't get another Expression Editor window, as I would expect. The new selected measure simply replaces the contents of the existing Expression Editor window. I realize I can go back and forth between my measures using the arrows in the upper right of the editor window but I really would expect another Expression Editor window to appear. In fact, I haven't found a way to actually get a second Expression Editor window to appear. Maybe I can't?

**Comment by otykier** [comment]
This is actually by design. The Expression Editor is intended for making quick changes to the expression of an object, hence why it only exists as a single instance. If you would like to have multiple DAX editors open, you should use the "Script DAX" option (right-click on the object > Script DAX). This opens a new window each time. Note that you can have multiple objects inside a single DAX script as well.

Feedback on this area is certainly welcome!

**Comment by rminder** [comment]
Thanks Daniel - I'm not certain this is what users would expect. It might
be helpful to think about how users are accustomed to creating their
measures and scripts in PBI. When I right-click on a measure in TE and
select Script DAX, the script I see is not what I am accustomed to seeing
in PBI. What I see is the full measure definition. I could be wrong but I
don't think most users would expect to work with their measures in this
way. I think most users would expect to pop up an instance of the
Expression Editor and start working on their measure, like in PBI. And be
able to have multiple Expression Editor windows open at once.

Just my $.02.

Randy
